lib/importer/src/index.js

```diff
@@ -477,6 +509,121 @@ exports.Initialise = (config, router, prototypeKit) => {
     },
   );
 
+  //--------------------------------------------------------------------
+  // Configuration management routes.
+  //--------------------------------------------------------------------
+
+  const TEMP_DIRECTORY_LABEL = "[Temporary directory]"
+
+  // Render the config page
+  router.get(
+    IMPORTER_ROUTE_MAP.get("importerConfiguration"),
+    (request, response) => {
+      const pc = {
+        config: {
+          fields: plugin_config.fields,
+          uploadPath: plugin_config.uploadPathDefault ? TEMP_DIRECTORY_LABEL : plugin_config.uploadPath
+        }
+      }
+
+      // Copy flash message to template object and then remove from memory
+      if (request.session.data['internal-message']) {
+        pc.message = request.session.data['internal-message']
+        request.session.data['internal-message'] = null
+      }
+
+      // This is a bit of a hacky way to render the error in the GET counterpart to a
+      // previous POST request at this URL.
+      if (request.session.data['internal-error']) {
+        pc.error_summary = request.session.data['internal-error'].summary
+        pc.error = request.session.data['internal-error'].error
+        pc.config.uploadPath = request.session.data['internal-error'].parameter
+        request.session.data['internal-error'] = null
+      }
+
+      response.status(200)
+      response.render("plugin_config.html", pc)
+    }
+  );
+
+
+  router.post(
+    IMPORTER_ROUTE_MAP.get("importerConfiguration"),
+    (request, response) => {
+      request.session.data['internal-message'] = "Configuration changes have been saved, and the application will restart shortly"
```

Comment:
Is it worth us redirecting to an interstitial page whilst its restarting, and then moving on after that?

Reply:
The moment the config page saves the new config, the app is restarting, so although we can show them a different page we wouldn't be able to automatically move on unless we relied on JS and a pause that we knew was longer than the restart time.